src/pages/home/HeaderView.js

```diff
@@ -49,66 +50,121 @@ const defaultProps = {
     report: null,
 };
 
-const HeaderView = (props) => {
-    const participants = lodashGet(props.report, 'participants', []);
-    const reportOption = {
-        text: lodashGet(props.report, 'reportName', ''),
-        tooltipText: getReportParticipantsTitle(participants),
-        participantsList: getPersonalDetailsForLogins(participants, props.personalDetails),
-    };
-
-    return (
-        <View style={[styles.appContentHeader]} nativeID="drag-area">
-            <View style={[styles.appContentHeaderTitle, !props.isSmallScreenWidth && styles.pl5]}>
-                {props.isSmallScreenWidth && (
-                    <Pressable
-                        onPress={props.onNavigationMenuButtonClicked}
-                        style={[styles.LHNToggle]}
-                    >
-                        <Icon src={BackArrow} />
-                    </Pressable>
-                )}
-                {props.report && props.report.reportName && (
-                    <View
-                        style={[
-                            styles.flex1,
-                            styles.flexRow,
-                            styles.alignItemsCenter,
-                            styles.justifyContentBetween,
-                        ]}
-                    >
+class HeaderView extends Component {
+    constructor(props) {
+        super(props);
+
+        this.participants = lodashGet(this.props.report, 'participants', []);
```

Comment:
Just thinking out loud: I think since you moved this from `render` to the `constructor`, these fields won't be updated if the props change. I think you would need to re-set them from `componentDidUpdate` for that to happen. But I also don't think that these props ever change after the component is created, so I think we're okay here. 🙂